### PR TITLE
Fixes two bugs in batch catchup

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

- Provide a description and a link to the underlying ticket

### What changes were made as part of this PR:

- Provide a high level list of the changes made

### Definition of done

- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated 
- [ ] End-to-end tests run if required (see below)

### End-to-end tests

Running the end-to-end tests in the [obscuro-test repo](https://github.com/obscuronet/obscuro-test) is currently at the 
discretion of the developer prior to merging a PR. The intention is not to slow down development by having the longer 
running end-to-end tests as a PR gate run on each branch commit etc. To manually trigger a run pre-merge;

- Go to [run_local_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_local_tests.yml)
- Click the "Run workflow" button
- Use the main branch of obscuro-test 
- Enter the name of the PR branch for go-obscuro
- Run the workflow

The run takes approximately 20 minutes and any failure will be reported in the workflow output. Should the tests fail 
the docker container output for the local testnet and all test artifacts will be added to the run and can be downloaded
for debugging and analysis, with a retention period of 2 days. 

Note that every PR merge to main will also trigger a run of post-merge tests, so even if you do not manually trigger 
pre-merge, tests will be run post-merge. The intention being to catch any issues fast on main to allow for quick 
resolution. The post-merge test output can be seen at 
[run_merge_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_merge_tests.yml) and will show both 
the PR number and author in the list of runs.

### CODE REVIEW

File: go/host/host.go [hunk 1]
```diff
@@ -875,7 +875,7 @@ func (h *host) handleBatches(encodedBatchMsg *common.EncodedBatchMsg) error {
 
 	for _, batch := range batchMsg.Batches {
 		// The enclave retrieves the genesis from the L1 chain, so we do not need to submit it.
-		if batch.Header.Number == big.NewInt(int64(common.L2GenesisHeight)) {
+		if batch.Header.Number.Cmp(big.NewInt(int64(common.L2GenesisHeight))) == 0 {
```
Comment (joeldudley) [inline go/host/host.go:878]: This meant that we'd never store the genesis batch, because we were doing the compare wrong.

File: go/host/host.go [hunk 2]
```diff
@@ -909,6 +909,7 @@ func (h *host) handleBatches(encodedBatchMsg *common.EncodedBatchMsg) error {
 				}
 				return nil
 			}
+			return nil
```
Comment (joeldudley) [inline go/host/host.go:912]: This meant that we were submitting batches even if the parent was missing.

File: go/common/retry/retry.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func Do(fn func() error, retryStrat Strategy) error {
 			return nil
 		}
 
-		var ffErr *failFastError
+		var ffErr *FailFastError
```
Comment (BedrockSquirrel) [inline go/common/retry/retry.go:21]: How come this had to be public? Did you hit some error from await receipt?

File: go/host/batchmanager/batch_manager.go [hunk 4]
```diff
@@ -66,7 +66,7 @@ func (b *BatchManager) GetBatches(batchRequest *common.BatchRequest) ([]*common.
 	if err != nil {
 		return nil, fmt.Errorf("could not determine latest canonical ancestor. Cause: %w", err)
 	}
-	batchesToSend := []*common.ExtBatch{firstBatch}
```
Comment (joeldudley) [inline go/host/batchmanager/batch_manager.go:69]: We need to add the first batch last, because we then reverse the list.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**